cmapPy/pandasGEXpress/concat_gctoo.py

```diff
@@ -262,50 +276,108 @@ def assemble_common_meta(common_meta_dfs, fields_to_remove):
         all_meta_df_sorted (pandas df)
 
     """
+    all_meta_df, all_meta_df_with_dups = build_common_all_meta_dfs(common_meta_dfs, fields_to_remove, remove_all_metadata_fields)
+
+    if not all_meta_df.index.is_unique:
+        all_report_df = build_mismatched_common_meta_report([x.shape for x in common_meta_dfs],
+            sources, all_meta_df, all_meta_df_with_dups)
+
+        unique_duplicate_ids = all_report_df.index.unique()
+
+        if error_report_file is not None:
+            all_report_df.to_csv(error_report_file, sep="\t")
+
+        msg = """There are inconsistencies in common_metadata_df between different files.  Try excluding metadata fields
+using the fields_to_remove argument.  unique_duplicate_ids: {}
+all_report_df:
+{}""".format(unique_duplicate_ids, all_report_df)
+        raise MismatchCommonMetadataConcatGctooException(msg)
+
+    # Finally, sort the index
+    all_meta_df_sorted = all_meta_df.sort_index(axis=0)
+
+    return all_meta_df_sorted
+
+
+def build_common_all_meta_dfs(common_meta_dfs, fields_to_remove, remove_all_metadata_fields):
```

Comment:
@dllahr just clarifying, since you didn't add docstrings (tsk tsk)-- I'm a little confused from the names of build_common_all_meta_dfs and build_mismatched_common_meta_report. My understanding is build_common_all_meta_dfs gets & outputs the intersection between two metadata dataframes (or a DF with just index if remove_all_metadata_fields = True), and that build_common_all_meta_dfs combines two metadata dfs of different dimensions. T/F? If not can you clarify? (or document 😄 )

Reply:
Self documenting code!!!  But you're write I should add a short synopsis of what these are doing.

Reply:
Yes please! I can't use self documenting code on ReadTheDocs.

Reply:
I'm not sure that it is critical to have documentation - especially ReadTheDocs - for "internal" methods like these - I created them because the assemble_common_meta method was getting too large, so they are just refactorings with some other minor changes / features added.  I do think it is important for code review though, so I've added them.